src/test/scala/mesosphere/marathon/api/ConstraintTest.scala

```diff
@@ -21,6 +22,20 @@ class ConstraintTest extends MarathonSpec {
     shouldMatch("""["jdk","UNLIKE","7"]""", "jdk", Constraint.Operator.UNLIKE, "7")
   }
 
+  test("Read should allow only valid Constraints (regression for #2951)") {
+    import mesosphere.marathon.api.v2.json.Formats._
+    Json.parse("""[]""").asOpt[Constraint] should be('empty)
```

Comment:
you can use `empty` instead of `'empty` with the advantage that it is type-checked.